### PR TITLE
Package the project with `nix flake init --template templates#rust`

### PR DESCRIPTION
**Pull request by yvan-sraka**

### CODE REVIEW

File: .envrc [hunk 1]
```diff
@@ -0,0 +1 @@
+use flake
```
Comment (sickill) [inline .envrc:1]: Won't this break direnv for people not using Nix?

File: .gitignore [hunk 2]
```diff
@@ -1 +1,3 @@
+.direnv
 /target
+result
```
Comment (sickill) [inline .gitignore:3]: What's in result? Nix creates it?

Comment (yvan-sraka) [reply .gitignore:3]: That's a symlink to the derivation produced by `nix build`!

File: .github/workflows/build_nix.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+name: "Build legacy Nix package on Ubuntu"
```
Comment (sickill) [inline .github/workflows/build_nix.yml:1]: One last thing: what's the purpose of this workflow? I guess this is only to test that it successfully builds with nix, right?

Comment (yvan-sraka) [reply .github/workflows/build_nix.yml:1]: Yes!

**Comment by sickill** [comment]
Great, merging. Thanks!

Note that I don't use Nix myself (yet!) and I can't really maintain this Nix config, so in case of any issues with this you will be pinged :) I hope that's ok.

**Comment by sickill** [comment]
@yvan-sraka  That ubuntu build fails: https://github.com/asciinema/agg/actions/runs/4589523213/jobs/8104469344#step:4:8

**Comment by yvan-sraka** [comment]
> @yvan-sraka That ubuntu build fails: https://github.com/asciinema/agg/actions/runs/4589523213/jobs/8104469344#step:4:8

I created a new PR that try to address the issue: https://github.com/asciinema/agg/pull/37

**Pull request merged**